primitives/src/pcs/mod.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Polynomial Commitment Scheme
+pub mod errors;
+mod multilinear_kzg;
+pub mod prelude;
+mod structs;
+mod transcript;
+mod univariate_kzg;
+
+use ark_ec::PairingEngine;
+use ark_serialize::CanonicalSerialize;
+use ark_std::{rand::RngCore, vec::Vec};
+use errors::PCSError;
+
+/// This trait defines APIs for polynomial commitment schemes.
+/// Note that for our usage of PCS, we do not require the hiding property.
+pub trait PolynomialCommitmentScheme<E: PairingEngine> {
+    /// Prover parameters
+    type ProverParam;
+    /// Verifier parameters
+    type VerifierParam;
+    /// Structured reference string
+    type SRS;
+    /// Polynomial and its associated types
+    type Polynomial;
+    /// Polynomial input domain
+    type Point;
+    /// Polynomial Evaluation
+    type Evaluation;
+    /// Commitments
+    type Commitment: CanonicalSerialize;
+    /// Batch commitments
+    type BatchCommitment: CanonicalSerialize;
+    /// Proofs
+    type Proof;
+    /// Batch proofs
+    type BatchProof;
+
+    /// Build SRS for testing.
+    ///
+    /// - For univariate polynomials, `supported_size` is the maximum degree.
+    /// - For multilinear polynomials, `supported_size` is the number of
+    ///   variables.
+    ///
+    /// WARNING: THIS FUNCTION IS FOR TESTING PURPOSE ONLY.
+    /// THE OUTPUT SRS SHOULD NOT BE USED IN PRODUCTION.
+    fn gen_srs_for_testing<R: RngCore>(
+        rng: &mut R,
```

Comment:
we should use marker trait `CryptoRng` (even tho it didn't affect anything, it's at least marking and indicating that users should pass in a cryptographic RNG.
i.e. change to `R: RngCore + CryptoRng`, similar to other occurrences.